session/session.go

```diff
@@ -478,9 +486,16 @@ func (s *session) doCommit(ctx context.Context) error {
 	if !s.txn.Valid() {
 		return nil
 	}
+
+	// to avoid session set overlap the txn set.
+	if s.GetDiskFullOpt() != kvrpcpb.DiskFullOpt_NotAllowedOnFull {
+		s.txn.SetDiskFullOpt(s.GetDiskFullOpt())
+	}
+
 	defer func() {
 		s.txn.changeToInvalid()
 		s.sessionVars.SetInTxn(false)
+		s.ClearDiskFullOpt()
```

Comment:
Reset the session setting after one transaction commits?

Reply:
Session flag setting is only used in and related with some specifiic operations.